charm_versions.md

```diff
@@ -0,0 +1,138 @@
+[channels](https://juju.is/docs/sdk/charmcraft-release).
+
+When a charm is deployed locally, Juju will use an initial revision of 1 and then increment the
+revision each time a new version of the same charm is deployed or refreshed.
+
+### Libraries
+
+Charm libs have both an "API version" and a "patch version". Any changes to the lib API that are not
+backwards compatible will be made in a new API version, while new patch versions include compatible
+improvements including new features, general improvements, and bug fixes.
+
+The available versions for a charm lib can be found on Charmhub in the "Libraries" tab. These can
+also be found using `charmcraft list-lib <charm name>`.
+
+The API and patch versions of a charm lib are contained within the lib source file itself. Search
+for `LIBAPI` (the API version) and `LIBPATCH` (the patch version) within the source.
+
+It is not possible to see which version of a lib was used in a deployed charm, other than by using
+the `charm-version` to examine the source tree from which the charm was built or retrieving the
+deployed charm source from a unit. In general, the lib version is a charm author concern, rather
+than one of an admin.
+
+Authors publish new versions of charm libraries using the
+[charmcraft publish-lib command](https://juju.is/docs/sdk/create-and-publish-a-charm-library#heading--update-a-charm-library),
+setting the [metadata](https://juju.is/docs/sdk/library#heading--structure) appropriately - in
+particular, the `LIBAPI` value for the API version (which is also in the path) and the `LIBPATCH`
+value for the patch version.
+
+To upgrade the revision of a lib used by a charm, the same `charmcraft fetch-lib` command is used
+as when originally installing the lib. To upgrade the version of the lib, use `fetch-lib` but
+change the version number in the path, for example from `charms.demo.v0.demo` to
+`charms.demo.v1.demo`. See the
+[documentation](https://juju.is/docs/sdk/find-and-use-a-charm-library#heading--update-a-charm-library)
+for more details.
+
+### Interfaces
+
+When integrating two charms together, it's important that both charms understand what the other
+side of the relation requires and provides. In the charm metadata this is declared using the
+[interface field of the relation](https://juju.is/docs/sdk/charmcraft-yaml#heading--peers-provides-requires).
+
+Interfaces are more meaningful when they are
+[registered](https://juju.is/docs/sdk/register-an-interface), which involves defining the data
+structures that each side of the relation will use in the relation databag, and creating a suite of
+tests that ensure that charms that require or provide the interface align with the specification.
+
+Interface versions are defined through the folder structure in the
+[charm-relation-interfaces repository](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/)
+and are updated by authors using pull requests.
+
```

Comment:
Interfaces should be documented at the level of the `interface` key in charmcraft.yaml: https://juju.is/docs/sdk/charmcraft-yaml#heading--peers-provides-requires (PS In this section, I intend to replace "relation type" and "relation name" with "endpoint role' and "endpoint name", to match the way we talk about this in Juju docs: https://juju.is/docs/juju/endpoint .) We may also decide to have a reference doc on the concept of an interface, but so far I am not convinced it is necessary.